go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/buildbuddy-io/buildbuddy
 
-go 1.21.7
+go 1.22.4
```

Comment:
This is needed to use the new `for range <int>` syntax.

We are already using Go 1.22, this is only raising the repo's minimum supported SDK version so that the Go language server and other static analysis won't complain.